docs/dev/adding_instruments.rst

```diff
@@ -327,3 +329,134 @@ The dictionary now maps the keys to specific values. The values and keys can be
+    )
+
+.. doctest::
+
+    >>> extreme = Extreme5000("GPIB::1")
+    >>> extreme.current = 1  # set current to 1000 mA
+
+Processing of return values
+***************************
+
+Similar to `set_process` the :func:`Instrument.control <pymeasure.instruments.Instrument.control>`, and :func:`Instrument.measurement <pymeasure.instruments.Instrument.measurement>` functions allow a `get_process` argument which if specified must be a function that takes a value and performs processing before value mapping. The function must return the processed value. In analogy to the example above this can be used for example for unit conversion:
+
+.. testcode::
+
+    Extreme5000.current = Instrument.control(
+        ":CURR?", ":CURR %g",
+        """ A floating point property representing the measurement current in A
+        """,
+        validator=strict_range,
+        values=[0, 10],
+        set_process=lambda v: 1e3*v  # convert to mA
+        get_process=lambda v: 1e-3*v  # convert to A
+    )
+
+.. doctest::
+
+    >>> extreme = Extreme5000("GPIB::1")
+    >>> extreme.current = 3.1
+    >>> extreme.current
+    3.1
+
+`get_process` can also be used to perform string processing. Let's say your instrument returns a value with its unit which has to be removed. This could be achieved by the following code:
+
+.. testcode::
+
+    Extreme5000.capacity = Instrument.measurement(
+        ":CAP?",
+        """ A measurement returning a capacity in nF in the format '<cap> nF'
+        """,
+        get_process=lambda v: float(v.replace('nF', ''))
+    )
+
+The same can be also achieved by the `preprocess_reply` keyword argument to :func:`Instrument.control <pymeasure.instruments.Instrument.control>` or :func:`Instrument.measurement <pymeasure.instruments.Instrument.measurement>`. This function is forwarded to :func:`Adapter.values <pymeasure.adapters.values>` and runs directly after receiving the reply from the device. One can therefore take advantage of the built in casting abilities and simplify the code accordingly:
+
+.. testcode::
+
+    Extreme5000.capacity = Instrument.measurement(
+        ":CAP?",
+        """ A measurement returning a capacity in nF in the format '<cap> nF'
+        """,
+        preprocess_reply=lambda v: v.replace('nF', '')
+    )
+
+The real purpose `preprocess_reply` is, however, for instruments where many/all properties need similar reply processing. `preprocess_reply` can be applied to all :func:`Instrument.control <pymeasure.instruments.Instrument.control>` or :func:`Instrument.measurement <pymeasure.instruments.Instrument.measurement>` properties. For example if all quantities are returned with a unit as in the example above. To avoid running into troubles for other properties this `preprocess_reply` should be clever enough to skip the processing in case for example some identification string is replied. Typically this can be achieved by regular expression matching. In case of no match the reply is returned unchanged:
```

Comment:
```suggestion
The real purpose of `preprocess_reply` is, however, for instruments where many/all properties need similar reply processing. `preprocess_reply` can be applied to all :func:`Instrument.control <pymeasure.instruments.Instrument.control>` or :func:`Instrument.measurement <pymeasure.instruments.Instrument.measurement>` properties, for example if all quantities are returned with a unit as in the example above. To avoid running into troubles for other properties this `preprocess_reply` should be clever enough to skip the processing in case it is not appropriate, for example if some identification string is returned. Typically this can be achieved by regular expression matching. In case of no match the reply is returned unchanged:
```